morse.js

```diff
@@ -62,14 +59,14 @@ window.Morse = (function () {
     return {
         encode: function (text) {
             return text.toUpperCase().split('')
-                .filter(letter => source.hasOwnProperty(letter))
+                .filter(letter => letter in source)
                 .map(letter => letterToMarkup(source[letter]).join(markup.space))
                 .join(byLetter())
         },
         decode: function (source) {
             return source
                 .split(byWord())
-                .map(word => word.split(byLetter()).filter(Boolean).map(letterFromMarkup).join(''))
+                .map(word => word.split(byLetter()).filter(String).map(letterFromMarkup).join(''))
```

Comment:
`Boolean` was added for delete empty elements in array, not sure `String` will do it